### PR TITLE
use remote theme

### PR DESCRIPTION
**Pull request by tobyhodges**
This should be everything you need to start using the remote theme. As well as minor changes to `.gitignore` and `_config.yml`, the main differences are:

1. structural information (`_layouts`, `_includes`, `assets` directories) are removed - after merging, this information will be drawn directly from the [remote theme repo](https://github.com/carpentries/carpentries-theme)
2. the `make serve` command, to build and serve pages locally, now uses Bundler (i.e. `bundle exec jekyll serve` rather than `jekyll serve` directly)

I have only tested this on my local machine and GitHub - maybe you'd like to pull this branch and confirm it works on your local setup before hitting the Big Green Button?

### CODE REVIEW

**Comment by jezcope** [comment]
Thanks @tobyhodges, this looks great and tidies things up nicely. I'll give it a try on my local  machine ASAP and let you know if I run into any troubles.

**Comment by tobyhodges** [comment]
Note: just updated the branch to also remove `.ruby-gemset` & `.ruby-version`

File: .gitignore [hunk 1]
```diff
@@ -3,9 +3,13 @@
...
 __pycache__
 _site
 .Rproj.user
 .Rhistory
 .RData
-.jekyll-cache
+.bundle/
+.vendor/
+.docker-vendor/
+Gemfile.lock
```
Comment (bencomp) [inline .gitignore:15]: Why would you ignore Gemfile.lock? `bundler` tells you to commit it, doesn't it, to enable reproducable builds?

File: Makefile [hunk 2]
```diff
@@ -1,39 +1,70 @@
...
+export SHELL = /bin/bash
+
 ## ========================================
 ## Commands for both workshop and lesson websites.
 
 # Settings
 MAKEFILES=Makefile $(wildcard *.mk)
-JEKYLL=jekyll
-JEKYLL_VERSION=3.7.3
+JEKYLL_VERSION=3.8.5
```
Comment (bencomp) [inline Makefile]: I understand from you that GitHub Pages uses v3.9. Should this match with Pages?

File: Makefile [hunk 3]
```diff
@@ -1,39 +1,70 @@
...
+JEKYLL=bundle install --path .vendor/bundle && bundle update && bundle exec jekyll
```
Comment (bencomp) [inline Makefile]: If I understand correctly, you are updating dependencies on each page build. I guess that is why you deleted the Gemfile.lock as well? Is this common practice for the Carpentries?

**Review by bencomp — APPROVED** [review]
I like the results very much, but have a few technical questions about using Jekyll. I've used Jekyll in different projects, but also committed the Gemfile.lock.

**Comment by tobyhodges** [comment]
I'm going to ping @fmichonneau to respond to those questions, as I'm sure he'll do a better job than I could. François is on leave until Tuesday, though, so I'm afraid you'll likely have to wait for definitive answers.

File: .gitignore (hunk 1 above)
Comment (jezcope) [reply .gitignore:15]: I agree, I'd usually commit `Gemfile.lock` as well as `Gemfile`, but I'll wait for @fmichonneau's comment as this seems to be how it's done in [carpentries/carpentries-theme](https://github.com/carpentries/carpentries-theme).

File: Makefile (hunk 2 above)
Comment (jezcope) [reply Makefile]: In fact the latest [`Makefile`](https://github.com/carpentries/styles/blob/gh-pages/Makefile) in the styles repository doesn't specify a specific Jekyll version but now uses a dedicated Carpentries docker image built for the purpose. This `Makefile` should probably be synchronised with that one.

Comment (tobyhodges) [reply Makefile]: Agreed, and good catch.

File: .gitignore (hunk 1 above)
Comment (fmichonneau) [reply .gitignore:15]: We don't commit the lock file because we always want to use the latest version of the `github-pages`.

**Comment by fmichonneau** [comment]
this looks good to me! Thank you all.

**Comment by tobyhodges** [comment]
Thanks François. I'll update the Makefile to match the latest version in carpentries/styles and then I think we're ready to go.

**Comment by tobyhodges** [comment]
The Makefile has been synced, so I think this is ready to be merged

**Comment by libcce** [comment]
Thanks @fmichonneau @tobyhodges!

**Pull request merged**